ci/check-protobuf-stability.sh

```diff
@@ -89,26 +93,30 @@ USAGE
   exit
   ;;
 --stable)
-  if [[ $TARGET == "main" ]]; then
-    # Check against the most recent stable tag.
-    # This check runs only on main or PRs targeting main.
-    #
-    # This check does not need to run on release branch commits because
-    # they are built sequentially, so no conflicts are possible and the per-PR
-    # check is enough.
-    readonly LATEST_STABLE_TAG="$(git tag | grep -v "snapshot" | sort -V | tail -1)"
-    # The v1.17 stable release includes the buf config file with the default name `buf.yml`.
-    # Starting with v1.18 we have multiple buf config files.
-    if [[ $LATEST_STABLE_TAG =~ "v1.17."* ]]; then
-      BUF_CONFIG_UPDATED=false
-    else
-      BUF_CONFIG_UPDATED=true
-    fi
-    BUF_GIT_TARGET_TO_CHECK=".git#tag=${LATEST_STABLE_TAG}"
-    check_protos
+  # Check against the most recent stable tag.
+  #
+  # This check does not need to run on release branch commits because
+  # they are built sequentially, so no conflicts are possible and the per-PR
+  # check is enough.
+  readonly RELEASE_BRANCH_REGEX="^release/.*"
+  LATEST_STABLE_TAG=""
+  if [[ "${TARGET}" =~ ${RELEASE_BRANCH_REGEX} ]]; then
+    readonly VERSION="${TARGET#release/}"
+    readonly VERSION_PREFIX="${VERSION%.x}"
+    readonly STABLE_TAGS=($(git tag | grep "v.*" | grep -v "snapshot" | sort -V))
+    LATEST_STABLE_TAG="$(for TAG in "${STABLE_TAGS[@]}"; do version_lte "${TAG#v}" "${VERSION_PREFIX}.999" && echo "$TAG" || :; done | tail -1)"
+  else
+    LATEST_STABLE_TAG="$(git tag | grep "v.*" | grep -v "snapshot" | sort -V | tail -1)"
```

Comment:
maybe as an extra safety check, check that the target is `main` here.

Reply:
Do we need it? The default is `main`, and even if it isn't/wasn't, there is probably no better option than comparing it with the latest stable release (also based on the fact that `--stable` has been used). Alternatively, I can handle the 3rd case and fail the script(?) WDYT, @cocreature?

Reply:
I was thinking of failing the script just to make sure we don’t silently do the wrong thing if we mess up the regex to match on the version branches.